### ISSUE TITLE
Blobfuse package for Ubuntu 22.04 LTS

### ISSUE THREAD

**Issue by jsmith-speedeon** [Azure/azure-storage-fuse#808]
### Which version of the blobfuse was used?
N/A

### Which OS (please include version) are you using?
Ubuntu 22.04 LTS

### What problem was encountered?
No package is currently available in the official Microsoft Jammy repository.

### Have you found a mitigation/solution?
Testing installing the old 20.04 LTS package manually instead of from the official microsoft repo seems to work. (With manual installation of libfuse2 from the default repos).

### By default, blobfuse logs errors to syslog. If this is relevant, is there anything in the syslog that might be helpful?
Entry in the cloud init log of the new VM reports blobfuse package not found.  Manual verification confirms it is not available from the Microsoft repository.

### If relevant, please share your mount command.

Requesting that a new package is built and distributed via the official Microsoft repository for Jammy.

**Comment by vibhansa-msft** [comment]
We do not release Blobfuse for Ubuntu-22. We are working on next-gen blobfuse, which is in preview mode as of now, and we have plans to release that on Ubuntu-22. For the time being you can use Ubuntu-20 package.

**Comment by andyzhangx** [comment]
> We do not release Blobfuse for Ubuntu-22. We are working on next-gen blobfuse, which is in preview mode as of now, and we have plans to release that on Ubuntu-22. For the time being you can use Ubuntu-20 package.

@vibhansa-msft is it possible still releasing blobfuse 1 for Ubuntu 22.04? AKS will support Ubuntu 22.04 very soon, and blobfuse2 is still in preview, I think there is a gap here.

**Comment by vibhansa-msft** [comment]
We are not taking any new feature request for v1, critical bug fixes only. If customer needs it work on ubn-22 as of now they have to manually build it for that environment. Blobfuse2 will support ubn-22 and GA is planned in Oct-Nov timeframe. So, the gap will be bridged soon.

**Comment by andyzhangx** [comment]
> We are not taking any new feature request for v1, critical bug fixes only. If customer needs it work on ubn-22 as of now they have to manually build it for that environment. Blobfuse2 will support ubn-22 and GA is planned in Oct-Nov timeframe. So, the gap will be bridged soon.

@vibhansa-msft since blobfuse v1 & v2 parameters are not compatible, when user upgrade agent node os version from ubn-18 to ubn-22, it would still break. It's ok that we only provide bug fixes for v1, while is it possible that we still provide an official way to install blobfuse v1 in ubn-18, the migration from v1 to v2 in AKS could take long time.

**Comment by vibhansa-msft** [comment]
Supporting a new distro requires whole cycle of testing and stabilizing v1 on that, which we do not intend to do as of now.
For the v1 to v2 migration, we already have "mountv1" command in v2, which takes exactly the same input and config file that user provides to v1 and automatically converts them to v2 compatible options and config file and mount the given directory using v2.
You can refer https://github.com/Azure/azure-storage-fuse/blob/main/MIGRATION.md for migration path.

**Comment by andyzhangx** [comment]
@vibhansa-msft we don't use config-file in k8s blob csi driver, we use pass all mount options in one command, here is an example: https://github.com/kubernetes-sigs/blob-csi-driver/blob/master/deploy/example/storageclass-blobfuse.yaml#L12-L21

**Comment by vibhansa-msft** [comment]
We do have cli options for v2 as well (just like v1). There are few gaps, but our preview-3 will have all these cli options and here is how the transformed options will look like:

mountv2Options:
  - --allow-other
  - --file-cache-timeout=120
  - --cancel-list-on-mount-seconds=10  # prevent billing charges on mounting
  - --attr-timeout=120
  - --entry-timeout=120
  - --negative-timeout=120
  - --log-level=LOG_WARNING  # LOG_WARNING, LOG_INFO, LOG_DEBUG
  - --cache-size-mb=1000  # Default will be 80% of available memory, eviction will happen beyond that.

**Comment by andyzhangx** [comment]
@vibhansa-msft thanks, what is `preview-3`? the main concern from blob csi driver side is that when migrating from v1 to v2, the cli options are not compatible, some options are only for v1 now, and if we migrate from our backend (e.g. during upgrade from 18.04 to 22.04), there could be potential issue.

**Comment by vibhansa-msft** [comment]
I agree some cli options have been renamed and a few are only available in v1. In 'blobfuse2' there is a 'mountv1' command which takes exactly the same cli options that v1 used to take and make it easier for customers to migrate to v2. If CSI driver starts using that command, there shall not be any breakages.  'preview-3' will be our latest blobfuse2 release, mostly coming out next week.
